src/test/ui/asm/type-check-2.rs

```diff
@@ -90,3 +90,11 @@ fn main() {
         asm!("{}", in(reg) u);
     }
 }
+
+// Const operands must be integer or floats, and must be constants.
+
+global_asm!("{}", const 0);
+global_asm!("{}", const 0i32);
+global_asm!("{}", const 0f32);
+global_asm!("{}", const 0 as *mut u8);
```

Comment:
There is a separate `sym` operand type for this which inserts a symbol name. It is not supported in `global_asm!` yet but will be in the future.

Reply:
Ah, I meant something different. Like

```
const GPIOA: *const u32 = 0x8000_4200;
```

and then users trying to use that pointer as-is. Symbol isn't quite usable here because placing the symbol in the right place would require linker shenanigans people don't generally love all that much.